### PR TITLE
[DROOLS-6663] Refactor kie-parent and create drools-build-parent

### PR DESCRIPTION
**Pull request by danielezonca**
https://issues.redhat.com/browse/DROOLS-6663

## Done
- Created `drools-build-parent` module that is the _real_ build root module
- Created `bom` module with all the Drools boms
- Made all root modules (except boms) declare this `drools-build-parent` as parent
- Removed a lot (_a lot_) of not-used or not necessary anymore dependencies and probably many more can be removed
- Removed root declaration of a global `junit`/`mockito-core` dependency and declare it everywhere is necessary
- Reintroduced jdk9+ profiles: I got NPE instead of proper error message in some case of missing test dependencies. I guess it could be because we have some mixed Maven dependencies but I'm not sure
- Removed usage of `version.org.kie` and replaced by `project.version`

## To be done
- ~~Solve remaining TODOs~~
- ~~Move properties to `build-parent`~~
- ~~Run script to rebase dependencies on Quarkus platform instead of current Wildfly~~
- ~~Check with CI to see if there are test errors~~ 🤞
- ~~Test if [DROOLS-6670](https://issues.redhat.com/browse/DROOLS-6670) is solved too (I expect so)~~
- ~~Finalise some cleanup of kie-core-bom~~
- ~~Fix some module name to fit the `Drools :: Naming :: Convention`~~
- ~~Fix dmn-gwt-showcase compilation~~

**Thank you for submitting this pull request**

**NOTE!:** Double check the target branch for this PR.
The default is `main` so it will target Drools 8 / Kogito / Optaplanner 8.
If this PR is not strictly related to drools and kogito project in `drools.git`, it should probably target `7.x`as a branch

**Ports** If a forward-port or a backport is needed, paste the forward port PR here

[link](https://www.example.com)

**JIRA**: _(please edit the JIRA link if it exists)_

[link](https://www.example.com)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>

* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>

* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: kie-dmn/kie-dmn-feel-gwt-showcase/pom.xml [hunk 1]
```diff
@@ -24,6 +24,28 @@
     <version.com.google.guava>29.0-jre</version.com.google.guava>
   </properties>
 
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.seleniumhq.selenium</groupId>
+        <artifactId>selenium-java</artifactId>
+        <version>${version.org.jboss.arquillian.selenium}</version>
```
Comment (danielezonca) [inline kie-dmn/kie-dmn-feel-gwt-showcase/pom.xml:32]: Try to import full selenium bom

**Comment by danielezonca** [comment]
@radtriste 
https://issues.redhat.com/browse/DROOLS-6670 is solved as part of this PR

**Comment by danielezonca** [comment]
@gitgabrio @tarilabs 
This PR includes a quite big cleanup of dependencies and configuration from "old" kie-parent for Drools8.
It has no impact in general on PMML and DMN but I fixed one "strange" configuration we have in 7 that had an impact on many poms: in 7 `junit` and `mockito` are declared as dependency in the root project (`kie-parent`) so each module has it by default. I removed this so now each module has to declare it properly (i.e. see [this change](https://github.com/kiegroup/drools/pull/3931/files#diff-d7558a9c11ef2b3f257f41c240ba6b230b886691393be5e646999c4d236b917c)).
I think we can backport the changes at DMN/PMML POMs level to 7 if necessary.
Wdyt?

**Review by mariofusco — APPROVED** [review]

**Comment by tarilabs** [comment]
Good with me @danielezonca thank you and yes I would agree it is worthy to backport to 7 👍

**Comment by danielezonca** [comment]
> Good with me @danielezonca thank you and yes I would agree it is worthy to backport to 7 👍

Just to be clear, I don't expect to backport the removal of `junit`/`mockito` from 7.x [kie-parent](https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/main/pom.xml#L6480-L6491) because this implies to check _all_ modules of 7.x in _all_ the repos and add the missing dependency but it is possible to backport this declaration for drools.git modules

**Review by radtriste — APPROVED** [review]
tested the change version and build locally and it works :)

**Review by gitgabrio — APPROVED** [review]
LGTM

**Comment by radtriste** [comment]
jenkins rerun drools tests

**Comment by radtriste** [comment]
jenkins rerun kogito-runtimes tests

**Comment by radtriste** [comment]
jenkins rerun kogito-examples tests

**Comment by radtriste** [comment]
jenkins rerun kogito-apps tests

**Comment by radtriste** [comment]
jenkins rerun drools tests

**Comment by radtriste** [comment]
for kogito-examples: https://github.com/kiegroup/kogito-examples/pull/951

**Comment by radtriste** [comment]
jenkins rerun kogito-runtimes tests

**Comment by radtriste** [comment]
jenkins rerun kogito-examples tests

**Comment by radtriste** [comment]
kogito-runtimes failure is specific to invoker and maven mirror which will be corrected in another PR. It should not impact the release

**Pull request merged**